

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Commands &mdash; Solosis 0.2.3 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="_static/css/custom.css?v=512ce168" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=a47416a6"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Examples" href="examples.html" />
    <link rel="prev" title="Setup" href="setup.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            Solosis
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Documentation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Commands</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#cli">cli</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#cli-alignment">alignment</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#cli-alignment-cellranger">cellranger</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cli-alignment-cellrangerarc">cellrangerARC</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cli-alignment-starsolo">starsolo</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#cli-filesystem">filesystem</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#cli-filesystem-disk-usage">disk-usage</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#cli-irods">irods</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#cli-irods-pull-cellranger">pull-cellranger</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cli-irods-pull-fastqs">pull-fastqs</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#cli-ncl-bsu">ncl-bsu</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#cli-ncl-bsu-migrate">migrate</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#cli-sc-rna">sc-rna</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#cli-sc-rna-cellbender">cellbender</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cli-sc-rna-merge-h5ad">merge-h5ad</a></li>
<li class="toctree-l4"><a class="reference internal" href="#cli-sc-rna-scanpy">scanpy</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="examples.html">Examples</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">More</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="input.html">Input File</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="development.html">Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">Run</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Project Links</span></p>
<ul>
<li class="toctree-l1"><a class="reference external" href="https://github.com/haniffalab/solosis">Source Code</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/haniffalab/solosis/issues">Issue Tracker</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Solosis</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Commands</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/commands.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="commands">
<h1>Commands<a class="headerlink" href="#commands" title="Link to this heading"></a></h1>
<section id="cli">
<h2>cli<a class="headerlink" href="#cli" title="Link to this heading"></a></h2>
<p>Command line utility for the Cellular Genetics
programme at the Wellcome Sanger Institute</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-version">
<span class="sig-name descname"><span class="pre">--version</span></span><span class="sig-prename descclassname"></span><a class="headerlink" href="#cmdoption-cli-version" title="Link to this definition"></a></dt>
<dd><p>Show the version and exit.</p>
</dd></dl>

<section id="cli-alignment">
<h3>alignment<a class="headerlink" href="#cli-alignment" title="Link to this heading"></a></h3>
<p>Commands for running alignment tools.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>alignment<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<section id="cli-alignment-cellranger">
<h4>cellranger<a class="headerlink" href="#cli-alignment-cellranger" title="Link to this heading"></a></h4>
<p>Run Cell Ranger for single-cell RNA sequencing alignment and analysis</p>
<p>Cell Ranger (7.2.0) performs sample demultiplexing, barcode processing,
and gene counting for single-cell 3’ and 5’ RNA-seq data, as well as
V(D)J transcript sequence assembly</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>alignment<span class="w"> </span>cellranger<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-cellranger-sample">
<span class="sig-name descname"><span class="pre">--sample</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;sample&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-cellranger-sample" title="Link to this definition"></a></dt>
<dd><p>Sample ID (string)</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-cellranger-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-cellranger-samplefile" title="Link to this definition"></a></dt>
<dd><p>Path to a CSV or TSV file containing sample IDs</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-cellranger-create-bam">
<span class="sig-name descname"><span class="pre">--create-bam</span></span><span class="sig-prename descclassname"></span><a class="headerlink" href="#cmdoption-cli-alignment-cellranger-create-bam" title="Link to this definition"></a></dt>
<dd><p>Generate BAM files for each sample</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-cellranger-version">
<span class="sig-name descname"><span class="pre">--version</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;version&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-cellranger-version" title="Link to this definition"></a></dt>
<dd><p>Cell Ranger version to use (e.g., ‘7.2.0’)</p>
</dd></dl>

</section>
<section id="cli-alignment-cellrangerarc">
<h4>cellrangerARC<a class="headerlink" href="#cli-alignment-cellrangerarc" title="Link to this heading"></a></h4>
<p>CellrangerARC aligns GEX &amp; ATAC seq reads…</p>
<hr class="docutils" />
<p>Cell RangerARC (2.0.2) Software suite designed for analysing &amp; interpreting scRNA seq data, including multi-omics data.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>alignment<span class="w"> </span>cellrangerARC<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-cellrangerARC-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-cellrangerARC-samplefile" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> Sample file text file</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-cellrangerARC-libraries">
<span class="sig-name descname"><span class="pre">--libraries</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;libraries&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-cellrangerARC-libraries" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> libraries file CSV</p>
</dd></dl>

</section>
<section id="cli-alignment-starsolo">
<h4>starsolo<a class="headerlink" href="#cli-alignment-starsolo" title="Link to this heading"></a></h4>
<p>STARsolo aligns single-cell RNA sequencing  reads…</p>
<p>STARsolo (2.7.11b) Aligner processes scRNA seq data to generate
GEX matrices &amp; identify cell-specific transcripts.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>alignment<span class="w"> </span>starsolo<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-starsolo-sample">
<span class="sig-name descname"><span class="pre">--sample</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;sample&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-starsolo-sample" title="Link to this definition"></a></dt>
<dd><p>Sample ID (string)</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-alignment-starsolo-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-alignment-starsolo-samplefile" title="Link to this definition"></a></dt>
<dd><p>Path to a CSV or TSV file containing sample IDs</p>
</dd></dl>

</section>
</section>
<section id="cli-filesystem">
<h3>filesystem<a class="headerlink" href="#cli-filesystem" title="Link to this heading"></a></h3>
<p>Commands for file and directory operations.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>filesystem<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<section id="cli-filesystem-disk-usage">
<h4>disk-usage<a class="headerlink" href="#cli-filesystem-disk-usage" title="Link to this heading"></a></h4>
<p>Check disk usage for a specified workspace.</p>
<p>This command retrieves the disk usage for a given workspace directory
across NFS, Lustre, or warehouse storage systems.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>filesystem<span class="w"> </span>disk-usage<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</section>
</section>
<section id="cli-irods">
<h3>irods<a class="headerlink" href="#cli-irods" title="Link to this heading"></a></h3>
<p>Commands for working with iRODS.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>irods<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<section id="cli-irods-pull-cellranger">
<h4>pull-cellranger<a class="headerlink" href="#cli-irods-pull-cellranger" title="Link to this heading"></a></h4>
<p>Downloading cellranger output(s) from iRODS..</p>
<p>This command retrieves cellranger output(s) generated by
the Sanger pipeline from iRODS storage.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>irods<span class="w"> </span>pull-cellranger<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-irods-pull-cellranger-sample">
<span class="sig-name descname"><span class="pre">--sample</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;sample&gt;</span></span><a class="headerlink" href="#cmdoption-cli-irods-pull-cellranger-sample" title="Link to this definition"></a></dt>
<dd><p>Sample ID (string)</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-irods-pull-cellranger-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-irods-pull-cellranger-samplefile" title="Link to this definition"></a></dt>
<dd><p>Path to a CSV or TSV file containing sample IDs</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-irods-pull-cellranger-retainbam">
<span class="sig-name descname"><span class="pre">--retainbam</span></span><span class="sig-prename descclassname"></span><a class="headerlink" href="#cmdoption-cli-irods-pull-cellranger-retainbam" title="Link to this definition"></a></dt>
<dd><p>Download alignment bam file</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-irods-pull-cellranger-overwrite">
<span class="sig-name descname"><span class="pre">--overwrite</span></span><span class="sig-prename descclassname"></span><a class="headerlink" href="#cmdoption-cli-irods-pull-cellranger-overwrite" title="Link to this definition"></a></dt>
<dd><p>Overwrite existing folder contents</p>
</dd></dl>

</section>
<section id="cli-irods-pull-fastqs">
<h4>pull-fastqs<a class="headerlink" href="#cli-irods-pull-fastqs" title="Link to this heading"></a></h4>
<p>Downloading fastqs from iRODS…</p>
<p>Utilising NF-irods-to-fastq pipeline developed by Cellgeni.
Pulled directly from Github repo- up-to-date.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>irods<span class="w"> </span>pull-fastqs<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-irods-pull-fastqs-sample">
<span class="sig-name descname"><span class="pre">--sample</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;sample&gt;</span></span><a class="headerlink" href="#cmdoption-cli-irods-pull-fastqs-sample" title="Link to this definition"></a></dt>
<dd><p>Sample ID (string)</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-irods-pull-fastqs-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-irods-pull-fastqs-samplefile" title="Link to this definition"></a></dt>
<dd><p>Path to a CSV or TSV file containing sample IDs</p>
</dd></dl>

</section>
</section>
<section id="cli-ncl-bsu">
<h3>ncl-bsu<a class="headerlink" href="#cli-ncl-bsu" title="Link to this heading"></a></h3>
<p>Commands for Newcastle University BSU.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>ncl-bsu<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<section id="cli-ncl-bsu-migrate">
<h4>migrate<a class="headerlink" href="#cli-ncl-bsu-migrate" title="Link to this heading"></a></h4>
<p>Placeholder</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>ncl-bsu<span class="w"> </span>migrate<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</section>
</section>
<section id="cli-sc-rna">
<h3>sc-rna<a class="headerlink" href="#cli-sc-rna" title="Link to this heading"></a></h3>
<p>Commands for single-cell RNA-seq workflows.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>sc-rna<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<section id="cli-sc-rna-cellbender">
<h4>cellbender<a class="headerlink" href="#cli-sc-rna-cellbender" title="Link to this heading"></a></h4>
<p>Cellbender (0.3.0.) Removes droplets &amp; ambient RNA from scRNA seq data.</p>
<hr class="docutils" />
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>sc-rna<span class="w"> </span>cellbender<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-sc-rna-cellbender-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-sc-rna-cellbender-samplefile" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> Sample file text file</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-sc-rna-cellbender-total_droplets_included">
<span class="sig-name descname"><span class="pre">--total_droplets_included</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;total_droplets_included&gt;</span></span><a class="headerlink" href="#cmdoption-cli-sc-rna-cellbender-total_droplets_included" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> total_droplets_included</p>
</dd></dl>

</section>
<section id="cli-sc-rna-merge-h5ad">
<h4>merge-h5ad<a class="headerlink" href="#cli-sc-rna-merge-h5ad" title="Link to this heading"></a></h4>
<section id="merging-multiple-h5ads-objects">
<h5>Merging multiple h5ads objects…<a class="headerlink" href="#merging-multiple-h5ads-objects" title="Link to this heading"></a></h5>
<p>Please run <cite>rna scanpy –samplefile …</cite> command first.</p>
<p>Example: /lustre/scratch126/cellgen/team298/soft/bin/examples/irods_download.txt
Input file should have 3 mandatory columns:
1st column: sanger_id, 2nd column: sample_name, LAST column: irods path</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>sc-rna<span class="w"> </span>merge-h5ad<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-sc-rna-merge-h5ad-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-sc-rna-merge-h5ad-samplefile" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> Sample file text file</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-sc-rna-merge-h5ad-merged_filename">
<span class="sig-name descname"><span class="pre">--merged_filename</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;merged_filename&gt;</span></span><a class="headerlink" href="#cmdoption-cli-sc-rna-merge-h5ad-merged_filename" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> Output file name: Eg. merged.h5ad</p>
</dd></dl>

</section>
</section>
<section id="cli-sc-rna-scanpy">
<h4>scanpy<a class="headerlink" href="#cli-sc-rna-scanpy" title="Link to this heading"></a></h4>
<section id="basic-scanpy-workflow-for-scrna-seq-data-generates-jupyter-notebook">
<h5>Basic Scanpy workflow for scRNA seq data, generates Jupyter Notebook …<a class="headerlink" href="#basic-scanpy-workflow-for-scrna-seq-data-generates-jupyter-notebook" title="Link to this heading"></a></h5>
<p>Example: /lustre/scratch126/cellgen/team298/soft/bin/examples/irods_download.txt
Input file should have 3 mandatory columns:
1st column: sanger_id, 2nd column: sample_name, LAST column: irods path</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cli<span class="w"> </span>sc-rna<span class="w"> </span>scanpy<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-sc-rna-scanpy-samplefile">
<span class="sig-name descname"><span class="pre">--samplefile</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;samplefile&gt;</span></span><a class="headerlink" href="#cmdoption-cli-sc-rna-scanpy-samplefile" title="Link to this definition"></a></dt>
<dd><p><strong>Required</strong> Sample file text file</p>
</dd></dl>

<dl class="std option">
<dt class="sig sig-object std" id="cmdoption-cli-sc-rna-scanpy-sample_basedir">
<span class="sig-name descname"><span class="pre">--sample_basedir</span></span><span class="sig-prename descclassname"> <span class="pre">&lt;sample_basedir&gt;</span></span><a class="headerlink" href="#cmdoption-cli-sc-rna-scanpy-sample_basedir" title="Link to this definition"></a></dt>
<dd><p>sample database folder</p>
</dd></dl>

</section>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="setup.html" class="btn btn-neutral float-left" title="Setup" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="examples.html" class="btn btn-neutral float-right" title="Examples" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Haniffa Lab.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>